oauth2/handler.go

```diff
@@ -682,6 +682,10 @@ func (h *Handler) AuthHandler(w http.ResponseWriter, r *http.Request, _ httprout
 	}
 
 	session, err := h.r.ConsentStrategy().HandleOAuth2AuthorizationRequest(w, r, authorizeRequest)
+	var obfuscatedSubject string
+	if err == nil {
+		obfuscatedSubject, err = h.r.ObfuscateSubjectIdentifier(authorizeRequest.GetClient(), session.ConsentRequest.Subject, session.ConsentRequest.ForceSubjectIdentifier)
+	}
```

Comment:
I suggest to move this after the `var accessTokenKeyID string` block just before https://github.com/ory/hydra/pull/2836/files#diff-dc5158991c1ce571f0a59c32b86d6fc64c55fbe11d67cdee44b1b0ae436569dfR727 with explicit error handling.